packages/cardano-services/test/util/polling/DbSyncEpochPollService.test.ts

```diff
@@ -0,0 +1,59 @@
+import { DbSyncEpochPollService } from '../../../src/util';
+import { InMemoryCache, UNLIMITED_CACHE_TTL } from '../../../src/InMemoryCache';
+import { Pool } from 'pg';
+import { ingestDbData, sleep, wrapWithTransaction } from '../../util';
+
+describe('DbSyncEpochPollService', () => {
+  const epochPollInterval = 2 * 1000;
+  const cache = new InMemoryCache(UNLIMITED_CACHE_TTL);
+  const db = new Pool({ connectionString: process.env.POSTGRES_CONNECTION_STRING, max: 1, min: 1 });
+  const epochMonitor = new DbSyncEpochPollService(db, epochPollInterval!);
+
+  describe('healthy state', () => {
+    afterAll(async () => {
+      await db.end();
+      cache.shutdown();
+    });
+
+    beforeEach(async () => {
+      cache.clear();
+    });
+
+    it(
+      'should execute all registered callbacks once the epoch rollover is detected by db polling',
+      wrapWithTransaction(async (dbConnection) => {
+        const currentEpoch = 205;
+        const greaterEpoch = 255;
+
+        const firstRegisteredCallback = jest.fn();
+        const secondRegisteredCallback = jest.fn();
+        const firstDisposer = epochMonitor.onEpochRollover(firstRegisteredCallback);
+        const secondDisposer = epochMonitor.onEpochRollover(secondRegisteredCallback);
+
+        await sleep(epochPollInterval * 2);
+
+        expect(await epochMonitor.getEpoch()).toEqual(currentEpoch);
+        expect(firstRegisteredCallback).not.toHaveBeenCalled();
+        expect(secondRegisteredCallback).not.toHaveBeenCalled();
+
+        await ingestDbData(
+          dbConnection,
+          'epoch',
+          ['id', 'out_sum', 'fees', 'tx_count', 'blk_count', 'no', 'start_time', 'end_time'],
+          [greaterEpoch, 58_389_393_484_858, 43_424_552, 55_666, 10_000, greaterEpoch, '2022-05-28', '2022-06-02']
+        );
+
+        await sleep(epochPollInterval * 2);
+
+        expect(firstRegisteredCallback).toHaveBeenCalled();
+        expect(secondRegisteredCallback).toHaveBeenCalled();
+        expect(await epochMonitor.getEpoch()).toEqual(greaterEpoch);
+        expect(cache.keys().length).toEqual(0);
```

Comment:
This test doesn't seem to be using `cache` for anything

```suggestion
```

Reply:
All the feedback is applied under https://github.com/input-output-hk/cardano-js-sdk/pull/370/commits/9c2bc2f0db9a36677df711c00f6dd5daf3a90a1a fixup commit @mkazlauskas